### DISCUSSION TITLE
Prevent scrolling to top when changing search params

### DISCUSSION THREAD

**Discussion by giopetris** [remix-run/remix#1918] [Proposals]
When using `<ScrollRestoration />` every query param change makes the page scroll to the top, but I guess we only want that when changing the **pathname**. 

I could get this tweak working locally by changing [this line](https://github.com/remix-run/remix/blob/1fd70960e4d88740df5bf407a6ba2cd2b9549459/packages/remix-react/scroll-restoration.tsx#L117) to `}, [location.pathname]);`, which then makes the callback to be executed only when the `pathname` changes. 

A common use case is a modal that opens through a query param for example, it doesn't make sense to scroll the entire page to the top when opening it. Or table results that are filtered based on the query params, it doesn't make sense to change the scroll position when changing the filters.

Does that change make sense? I'm happy to open a PR if so.

**Comment by ccssmnn** [comment]
I would love this!

**Comment by arnaudambro** [comment]
well in my code, I update the query params and it doesn't scroll to top

can you share a bit of your code to see what could be wrong ?

**Comment by giopetris** [comment]
After upgrading to remix 1.2.2 it stop happening, although, I'd like to update the query string without making a request to the server - if I put a `console.log` in the loader of that route it executes whenever there's a change in the query string, is there a way to prevent that? 

For now I'm using `history.replaceState();` but it would be great if remix or react-router has something built-in.

**Comment by giopetris** [comment]
Continue discussion in https://github.com/remix-run/remix/discussions/1920
